buildSrc/src/main/groovy/org/elasticsearch/gradle/precommit/NamingConventionsTask.groovy

```diff
@@ -111,16 +111,27 @@ public class NamingConventionsTask extends LoggedExec {
                 if (':build-tools'.equals(project.path)) {
                     args('--self-test')
                 }
+
                 if (checkForTestsInMain) {
                     args('--main')
                     args('--')
-                    args(project.sourceSets.main.output.classesDir.absolutePath)
                 } else {
                     args('--')
-                    args(project.sourceSets.test.output.classesDir.absolutePath)
                 }
+                args(getJavaClassesDir().absolutePath)
             }
         }
         doLast { successMarker.setText("", 'UTF-8') }
     }
+
+    File getJavaClassesDir() {
+        FileCollection classesDirs = (checkForTestsInMain ?
+                project.sourceSets.main.output.classesDirs :
+                project.sourceSets.test.output.classesDirs)
+        if (classesDirs.isEmpty()) { return null }
+        // see https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceSetOutput.html the deprecated property
+        // classesDir is the first in the FileCollection. There seems to be no other way to refer to the Java output dir
+        // We can't run the check against all classes as Groovy tests don't adhere to naming conventions.
+        return classesDirs[0]
```

Comment:
Why don't we change the java code to work on multiple directories? Then you can just pass in all the classesDirs as multiple arguments.

Reply:
Where do we have have groovy tests? How do we know the ordering would always result in the java classes dir coming first?

Reply:
we have groovy  `./buildSrc/src/test/groovy/`. It's an odd that the Gradle API does not offer a more explicit way to get the java specific dirs, and choose to document that it will always be first here: 
https://docs.gradle.org/current/dsl/org.gradle.api.tasks.SourceSetOutput.html 
It mentions the following for the deprectaed `File classesDir`  property:
```
Default with java plugin:
First directory from classesDirs
```

Reply:
While that deprecated property may document that fact, there would be nothing preventing gradle from changing the order once that property is removed. There are only 2 groovy tests. I think we can update their names to follow the naming convention?

Reply:
@rjernst I updated the task to pass along all dirs, but also added a new input to be able to bypass that and be specific of the  directory. Adding the Groovy tests is trickier because of the inheritance check but we'll be ok, since the build tools that have a hard coded path now are self tests, so if the test would change the test would fail.